vscode/microsoft-kiota/src/util.ts

```diff
@@ -162,15 +162,24 @@ export function allGenerationLanguagesToString(): string[] {
   return allSupportedLanguages;
 }
 
-export function validateDeepLinkQueryParams(queryParameters: Record<string, string>):
- [Record<string, string|undefined>, string[]]
+export interface IntegrationParams {
+  descriptionurl: string;
+  name: string;
+  kind: string;
+  type: string;
+  language: string;
```

Comment:
These two fields can have a string union type so that it matches with the intended contract with TTK
     "type":"ApiPlugin|ApiManifest|OpenAI", //Only useful when kind is plugin
       "language": "csharp|python|go|typescript|java|php|ruby|swift|cli"
```suggestion
  type: 'ApiPlugin' | 'ApiManifest' | 'OpenAI';
  language: 'csharp' | 'python' | 'go' | 'typescript' | 'java' | 'php' | 'ruby' | 'swift' | 'cli';
```

Reply:
These values are dynamic and may change. So best to rely on the available enums to parse the values. Hardcoding them would remove the benefit of single point of reference.